lib/rucio/rse/protocols/protocol.py

```diff
@@ -45,7 +45,7 @@ def __init__(self, protocol_attr, rse_settings):
             if getattr(rsemanager, 'SERVER_MODE', None):
                 setattr(self, '_get_path', self._get_path_nondeterministic_server)
         else:
-            self.attributes['determinism_type'] = 'default'
+            self.attributes['determinism_type'] = self.rse.get('determinism_type', 'default')
```

Comment:
Need to check if `determinsm_type` is part of the rse_settings. I am not sure.

Reply:
it's deterministic, it is defined on rse and should be in rse_seetings

Reply:
Yes, but deterministic is a boolean flag of the RSE. So for this you would need to have both the flag plus the determinism_type.